src/graphql/client_graph_account.rs

```diff
@@ -0,0 +1,78 @@
+use crate::{graphql::QueryError, Account};
+use graphql_client::{GraphQLQuery, Response};
+
+/// Derived GraphQL Query to Network Subgraph
+#[derive(GraphQLQuery)]
+#[graphql(
+    schema_path = "src/graphql/schema_graph_account.graphql",
+    query_path = "src/graphql/query_graph_account.graphql",
+    response_derives = "Debug, Serialize, Deserialize"
+)]
+pub struct GraphAccount;
+
+/// Query network subgraph for Network data
+/// Contains indexer address, stake, allocations
+/// and graph network minimum indexer stake requirement
+pub async fn query_graph_account(
+    url: String,
+    operator: String,
+    account: String,
+) -> Result<Account, QueryError> {
+    // Can refactor for all types of queries
+    let variables: graph_account::Variables = graph_account::Variables {
+        // Do not supply operator address if operator is already a graph account
+        operator_addr: if operator.clone() == account.clone() {
+            vec![]
+        } else {
+            vec![operator.clone()]
+        },
+        account_addr: account.clone(),
+    };
+    let request_body = GraphAccount::build_query(variables);
+    let client = reqwest::Client::builder()
+        .user_agent("network-subgraph")
+        .build()?;
+    let request = client.post(url.clone()).json(&request_body);
+    let response = request.send().await?.error_for_status()?;
+    let response_body: Response<graph_account::ResponseData> = response.json().await?;
+
+    if let Some(errors) = response_body.errors.as_deref() {
+        let e = &errors[0];
+        if e.message == "indexing_error" {
+            return Err(QueryError::IndexingError);
+        } else {
+            return Err(QueryError::Other(anyhow::anyhow!("{}", e.message)));
+        }
+    }
+    let data = if let Some(data) = response_body.data {
+        data
+    } else {
+        return Err(QueryError::ParseResponseError(format!(
```

Comment:
I couldn't figure it out quickly, but there must be a way to convert all occurrences of `return Err(QueryError ...)` to use `?`.

It isn't a must for this review, but it would make things simpler for you guys :)

Reply:
I figured a slightly simpler way but maybe it could be even better
```
    let data = response_body.data.map_or(
        Err(QueryError::ParseResponseError(format!(
        "Missing response data from network subgraph for account {account} with agent {operator}"
    ))),
        |data| Ok(data),
    )?;
```

Reply:
this seems to be better, will update to this :)
```
    let data = response_body
    .data
    .ok_or_else(|| QueryError::ParseResponseError(format!("Missing response data from network subgraph for account {} with agent {}", account, operator)))?;
```